tests/tests_integration/test_logging.py

```diff
@@ -0,0 +1,404 @@
+        )
+        log_filepath = log_filepath[0]
+
+        with open(log_filepath, "r") as file:
+            log = file.read()
+
+        return log
+
+    def delete_log_files(self, logging_path):
+        ds_logger.close_log_filehandler()
+        logs = glob.glob((str(logging_path / "*.log")))
+        for log in logs:
+            os.remove(log)
+
+    def test_logs_make_config_file(self, clean_project_name, tmp_path):
+        """"""
+        project = DataShuttle(clean_project_name)
+
+        project.make_config_file(
+            tmp_path, "two", "local_filesystem", use_behav=True
+        )
+
+        log = self.read_log_file(project._logging_path)
+
+        assert "Starting make_config_file" in log
+        assert "Successfully created rclone config." in log
+        assert (
+            "Configuration file has been saved and options loaded into datashuttle."
+            in log
+        )
+        assert "Update successful. New config file:" in log
+
+    def test_logs_update_config(self, setup_project):
+
+        setup_project.update_config("remote_path", "test_path")
+
+        log = self.read_log_file(setup_project._logging_path)
+
+        assert "Starting update_config" in log
+        assert "remote_path has been updated to test_path" in log
+        assert "Update successful. New config file:" in log
+        assert """ "remote_path": "test_path",\n """ in log
+
+    def test_logs_supply_config(self, setup_project, tmp_path):
+        """"""
+        new_configs_path, __ = test_utils.make_correct_supply_config_file(
+            setup_project, tmp_path
+        )
+        self.delete_log_files(setup_project._logging_path)
+        orig_project_path = setup_project._logging_path
+
+        setup_project.supply_config_file(new_configs_path, warn=False)
+
+        log = self.read_log_file(orig_project_path)
+
+        assert "supply_config_file" in log
+        assert "Update successful. New config file: " in log
+        assert (
+            f""" "local_path": "{setup_project.cfg['local_path'].as_posix()}",\n """
+            in log
+        )
+
+    def test_make_sub_dir__(self, setup_project):
+
+        subs = ["sub-1_1", "sub-002@TO@004"]
+        ses = ["ses-123", "ses-hello_world"]
+
+        setup_project.make_sub_dir(subs, ses, data_type="all")
+
+        log = self.read_log_file(setup_project._logging_path)
+
+        assert "Formatting Names..." in log
+        assert "sub_names: ['sub-1_1', 'sub-002@TO@004']" in log
+        assert "ses_names: ['ses-123', 'ses-hello_world']" in log
+        assert (
+            "formatted_sub_names: ['sub-1_1', 'sub-002', 'sub-003', 'sub-004']"
+            in log
+        )
+        assert "formatted_ses_names: ['ses-123', 'ses-hello_world']" in log
+        assert "Made directory at path:" in log
+        assert (
+            os.path.join("test_logging", "local", "rawdata", "sub-1_1") in log
+        )
+        assert (
+            os.path.join(
```

Comment:
I know it would need to be converted to a string at the end, but I would use Pathlib here for consistency. Mostly because familiarity with os.path will probably decline over time.